provider/pkg/provider/provider.go

```diff
@@ -1372,13 +1372,16 @@ func (k *kubeProvider) Diff(
 	if metadata.ReplaceUnready(newInputs) {
 		switch newInputs.GetKind() {
 		case "Job":
-			jobChecker := states.NewJobChecker()
-			job, err := clients.FromUnstructured(newInputs)
-			if err == nil {
-				jobChecker.Update(job)
-				if !jobChecker.Ready() {
-					hasChanges = pulumirpc.DiffResponse_DIFF_SOME
-					replaces = append(replaces, `.metadata.annotations["pulumi.com/replaceUnready"]`)
+			// Fetch current Job status and check point-in-time readiness. Errors are ignored.
+			if live, err := k.readLiveObject(newInputs); err == nil {
```

Comment:
Extending from @lukehoban's comment here: https://github.com/pulumi/pulumi-kubernetes/pull/1575#discussion_r632209070
should we be more fail-fast with error handling in providers in general? We don't have very good means of surfacing "warning" type diagnostics to the CLI so I wonder if we should bubble up the errors here rather than swallow them and result in surprising behavior for the user?

Reply:
In this particular case, the fallback behavior is a regular update, which would fail in the usual manner in case of a persistent error. Since this is more conservative than replacing, I'm inclined to just ignore the error.